src/main/java/com/andrewthesailor/cinemaTickets/service/impl/TicketTypeServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package com.andrewthesailor.cinemaTickets.service.impl;
+
+import com.andrewthesailor.cinemaTickets.DAO.TicketTypeDAO;
+import com.andrewthesailor.cinemaTickets.exception.TicketException;
+import com.andrewthesailor.cinemaTickets.model.TicketType;
+import com.andrewthesailor.cinemaTickets.service.TicketTypeService;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.NoResultException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class TicketTypeServiceImpl implements TicketTypeService {
+
+    final
+    TicketTypeDAO ticketTypeDAO;
+
+    @Override
+    public TicketType getByCode(String code) throws NoResultException {
+        return ticketTypeDAO.getByCode(code);
+    }
+
+
+    @Override
+    public void verifyCodes(List<String> ticketTypeCodes) {
+        List<String> codes = ticketTypeDAO.getCodes();
```

Comment:
Czy to nie powinno być jakoś cachowane? Wygląda na wolnozmienne.

Reply:
Gdybym jeszcze posiedział nad tym kodem, żeby go doszlifować to pewnie dorzuciłbym cache, w sumie to z 2 adnotacje.